cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDataset.java

```diff
@@ -85,12 +91,100 @@ public BusinessMetadataRecord getBusinessMetadata(Id.NamespacedId targetId, Stri
       return null;
     }
 
-    byte[] keyvalue = row.get(KEYVALUE_COLUMN);
     byte[] value = row.get(VALUE_COLUMN);
 
     return new BusinessMetadataRecord(targetId, key, Bytes.toString(value));
   }
 
+  /**
+   * Retrieves the business metadata for the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId the specified {@link Id.NamespacedId}
+   * @return a Map representing the metadata for the specified {@link Id.NamespacedId}
+   */
+  public Map<String, String> getBusinessMetadata(Id.NamespacedId targetId) {
```

Comment:
So metadata is a map of string key to string value. The only complex structure in the value is that for the 'tags' key, we have a comma-separated string. We do not allow to store any other data structures at this point, which is why I kept it as ``Map<String, String>``